src/grid.py

```diff
@@ -92,30 +131,26 @@ def rotate_turbine_locations(coords: List[Vec3], wd) -> Tuple[List[float], List[
         y_coord_rotated = x_coord_offset * sind(angle) + y_coord_offset * cosd(angle) + y_center_of_rotation
         return x_coord_rotated, y_coord_rotated
 
-    @property
-    def n_turbines(self) -> int:
-        return len(self.turbine_coordinates)
-
 
+@attr.s(auto_attribs=True)
 class TurbineGrid(Grid):
-    def __init__(
-        self,
-        turbine_coordinates: List[Vec3],
-        reference_turbine_diameter: float,
-        wind_directions: np.array,
-        wind_speeds: np.array,
-        grid_resolution: int,
-    ) -> None:
-        # establishes a data structure with grid on each turbine
-        # the x,y,z points here are the grid points on the turbine swept area
-        # TODO: should reference_turbine_diameter be here or in Farm? currently in both.
-        #       Both may be fine, they could be different maybe.
-        super().__init__(
-            turbine_coordinates,
-            reference_turbine_diameter,
-            grid_resolution,
-        )
-        self.set_grid(wind_directions, wind_speeds)
```

Comment:
We need to pass `wind_directions` and `wind_speeds` as arguments to the init. How should that be handled with attrs? We need the subclass to have additional arguments than those required by the super.

Reply:
This is just out of date with the latest PR, I merged in your changes but didn't check all the functionality thoroughly after merging, I'll fix this one.